File: programacao/c/respostas/selecao/17.c
#include <stdio.h>

int main(void) {
    int compra, pago, troco, cedulas;

    printf("Digite o valor da compra: ");
    scanf("%d", &compra);
    printf("Digite o valor pago: ");
    scanf("%d", &pago);

    troco = pago - compra;

    cedulas = troco / 100;
    if (cedulas > 0) {
        printf("%d cedula(s) de 100\n", cedulas);
        troco = troco % 100;
    }

    cedulas = troco / 50;
    if (cedulas > 0) {
        printf("%d cedula(s) de 50\n", cedulas);
        troco = troco % 50;
    }

    cedulas = troco / 20;
    if (cedulas > 0) {
        printf("%d cedula(s) de 20\n", cedulas);
        troco = troco % 20;
    }

    cedulas = troco / 10;
    if (cedulas > 0) {
        printf("%d cedula(s) de 10\n", cedulas);
        troco = troco % 10;
    }

    cedulas = troco / 5;
    if (cedulas > 0) {
        printf("%d cedula(s) de 5\n", cedulas);
        troco = troco % 5;
    }

    cedulas = troco / 2;
    if (cedulas > 0) {
        printf("%d cedula(s) de 2\n", cedulas);
        troco = troco % 2;
    }

    cedulas = troco / 1;
    if (cedulas > 0) {
        printf("%d cedula(s) de 1\n", cedulas);
    }

    return 0;
}

File: programacao/python/respostas/selecao/17.py
compra = int(input("Digite o valor da compra: "))
pago = int(input("Digite o valor pago: "))

troco = pago - compra

cedulas = troco // 100
if cedulas > 0:
    print(f"{cedulas} cedula(s) de 100")
    troco = troco % 100

cedulas = troco // 50
if cedulas > 0:
    print(f"{cedulas} cedula(s) de 50")
    troco = troco % 50

cedulas = troco // 20
if cedulas > 0:
    print(f"{cedulas} cedula(s) de 20")
    troco = troco % 20

cedulas = troco // 10
if cedulas > 0:
    print(f"{cedulas} cedula(s) de 10")
    troco = troco % 10

cedulas = troco // 5
if cedulas > 0:
    print(f"{cedulas} cedula(s) de 5")
    troco = troco % 5

cedulas = troco // 2
if cedulas > 0:
    print(f"{cedulas} cedula(s) de 2")
    troco = troco % 2

cedulas = troco // 1
if cedulas > 0:
    print(f"{cedulas} cedula(s) de 1")
